guidance/models/_model.py

```diff
@@ -1364,6 +1371,9 @@ def _run_stateless(self, stateless_function, temperature=0.0, top_p=1.0, n=1):
         # we will return a new extended version of ourselves, which we track as `lm`
         lm = self
 
+        # Prepare our metrics update. This is part of our Thread Unsafety programme
+        metrics_before = lm.engine.metrics.model_copy(deep=True)
```

Comment:
I'm not a huge fan of this, but otherwise I seemed to be getting double counting within the `for chunk in gen_obj` a few lines later.